packages/graphql/src/connections/Elasticsearch/ElasticsearchSearch/ElasticsearchSearch.js

```diff
@@ -0,0 +1,101 @@
+/*
+  Copyright 2020-2021 Lowdefy, Inc
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { Client } from '@elastic/elasticsearch';
+import schema from './ElasticsearchSearch.json';
+
+async function elasticsearchSearch( { request, connection } ) {
+    const client = new Client( connection );
+
+    // Wrap the query in a query property, if necessary
+    const body = 'query' in request.query
+                 ? request.query
+                 : { query: request.query };
+
+    if ( request.aggs ) {
+        body.aggs = { ...( body.aggs || {} ), ...request.aggs };
+    }
+
+    // Extract all parameters except "query"
+    const params = Object
+        .entries( request )
+        .filter( ( [ key ] ) => ![ 'query', 'aggs' ].includes( key ) )
+        .reduce( ( params, [ key, value ] ) => ( {
+            ...params,
+            [ key ]: value,
+        } ), {} );
+
+    const response = await client.search( {
+        ...params,
+        index: request.index || connection.index,
+        body,
+    } );
+
+    // Use the actual documents from the response as the base return value, then
+    // augment it with the additional, potentially interesting props.
+    const hits = response.body?.hits?.hits || [];
```

Comment:
@Gervwyk, I am not sure if our current build setup with babel targeting node 12 supports optional chaining. We have used the `get` function from `@lowdefy/helpers` for this in the past, but it might be better switch to optional chaining.

Reply:
We've been wanting to switch get for optional chaining for a while. Think it would make sense to switch everything at once to keep the code base consistent. Should we add it as a chore for v4?

Reply:
Yes, we've been wanting to switch everything to optional chaining for a while. Maybe it is best to do it all at once for consistency in the codebase. Let's add it at a chore for v4?

Reply:
Ah, sorry! I'm so used to optional chaining by now, didn't think about support. I can swap them out for the helper if that's easier for now?

Reply:
Optional chaining is great. From a consistency point of view, let's use [get](https://github.com/lowdefy/lowdefy/blob/main/packages/helpers/src/get.js), you'll see we use it everywhere, then when we do #680 we'll convert it everywhere.

Reply:
Fixed in bba533a